backend/main.cpp

```diff
@@ -0,0 +1,199 @@
+// *********************************************************************
+// ||                            INCLUDES                             ||
+// *********************************************************************
+#include <iostream>
+#include <boost/asio.hpp>
+#include <cstdint>
+#include "include/nlohmann/json.hpp"
+#include "matrix.hpp"
+
+using namespace boost::asio;
+
+// *********************************************************************
+// ||                             DEFINES                             ||
+// *********************************************************************
+#define POST_PORT 0xB00B
+
+// *********************************************************************
+// ||                            FUNCTIONS                            ||
+// *********************************************************************
+// AI did this, I have no idea if this is legit -Pat
+void handle_options_request(ip::tcp::socket& socket) {
+    // Send CORS headers for preflight request
+    std::string response = "HTTP/1.1 200 OK\r\n";
+    response += "Access-Control-Allow-Origin: *\r\n";
+    response += "Access-Control-Allow-Methods: DELETE, POST, GET, OPTIONS\r\n";
+    response += "Access-Control-Allow-Headers: Content-Type, Authorization, X-Requested-With\r\n";
+    response += "Content-Length: 0\r\n\r\n";
+    boost::asio::write(socket, boost::asio::buffer(response));
+}
+
+void handle_request(const std::string& request, ip::tcp::socket& socket) {
+    try {
+
+        std::cout << "Request received: " << request << std::endl;
+        // Handle OPTIONS request
+        if (request.substr(0, 7) == "OPTIONS") {
+            handle_options_request(socket);
+            return;
+        }
+
+        else if (request.substr(0, 4) != "POST") {
+            // Send a 405 Method Not Allowed response
+            std::string response = "HTTP/1.1 405 Method Not Allowed\r\n";
+            response += "Content-Type: text/plain\r\n";
+            response += "Access-Control-Allow-Origin: *\r\n"; // Enable CORS for all origins
+            response += "Content-Length: 0\r\n\r\n";
+            boost::asio::write(socket, boost::asio::buffer(response));
+            return;
+        }
+
+        if (request.empty()) {
```

Comment:
I dont think this line is reachable with an empty request, cause req == "" != "POST", I dont actually care if we handle it cause it's just gonna be us sending api requests, but just making notes if we want to organize the code later